src/py/flwr/server/superlink/fleet/grpc_rere/server_interceptor.py

```diff
@@ -133,6 +123,33 @@ def _generic_method_handler(
                         ),
                     )
                 )
+                try:
+                    node_id_from_client_public_key = self.state.get_node_id(
+                        client_public_key_bytes
+                    )
+                except KeyError:
+                    node_id_from_client_public_key = None
+
+                if node_id_from_client_public_key is not None:
+                    self.state.restore_node(
+                        node_id_from_client_public_key, request.ping_interval
+                    )
+                    self.state.store_node_id_client_public_key_pair(
+                        client_public_key_bytes, node_id_from_client_public_key
+                    )
```

Comment:
Why does this need to be stored again? Wasn't it stored before already?

Reply:
Removed.